src/robotFactory.js

```diff
@@ -1,10 +1,101 @@
 'use strict';
 
-class BaseRobot {}
+class Coords {
+  constructor() {
+    this.x = 0;
+    this.y = 0;
+  }
+}
 
-class FlyingRobot {}
+class BaseRobot {
+  constructor(
+    name,
+    weight,
+    coords,
+    chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class DeliveryDrone {}
+    this.coords
+      = {
+        ...new Coords(),
+        ...coords,
+      };
```

Comment:
this does not really make sense
Coords seems a bit redundant

if you want to make use of it, make it accept parameters and return an object with them, or return default values and simply do

```suggestion
    this.coords = new Coords(coords);
```

also, `never` brake line before or after `=`

Reply:
I used this in the case when `coords` it cames with a single parameter, like `{ x: 5 }` and I need to add `{ y: 0 }`...